### ISSUE TITLE
Support Foundry v9

### ISSUE THREAD

**Issue by caewok** [felddy/foundryvtt-docker#231]
Please add support for Foundry V9 Prototype 1. I understand that Foundry recently re-did its versioning, so this might require some adjustments to how you have been tagging releases.

FYI, I tried to side load V9 by adding it to container_caches and adding `FOUNDRY_VERSION=0.9.220` to my docker compose file. This almost works. I can see in the logs:
> Foundry Virtual Tabletop - Version 9

The license screen shows up and I can confirm it is running v9. But after confirming the license agreement, I get admin authentication successful and then instead of the setup screen, I see a blank Foundry screen with no way to continue. Logs show:
> Software license verification succeeded
> Administrator authentication successful for session [session id]
> global.ver is deprecated since V9 [repeated if I try re-loading the page]

Thanks for all your work on this container! Really very useful.

**Comment by felddy** [comment]
Thank you for opening the issue.  I'm glad the container is helpful to you.  

I've started work on this already and hit the same problem.  I've determined there is a bug in FoundryVTT v9.220 and opened an issue on GitLab to track it:

https://gitlab.com/foundrynet/foundryvtt/-/issues/5693

The new versioning scheme used by FoundryVTT will be a bit of a chore to work around, but I think I've come up with a good strategy.  I'm probably not going to bend over backwards to tag the myriad phases of their development cycle.  There will probably be "release" and "pre-release" containers.  

I'll push up my current progress to a branch when I get a chance, and disable the `--noupdate` flag that is currently malfunctioning.  This should get you up and running with an `edge` container.

**Comment by felddy** [comment]
I've pushed the branch I am working on up.  Unfortunately another casualty of the ["new approach to versioning"](https://foundryvtt.com/article/versioning/) has snagged testing in CI.  Foundry is reporting that it is version `9.0.0` which causes the pre-installed images to fail during testing.  

I've opened another issue on GitLab to request clarification or remediation of the `version` field in `package.json`: 
https://gitlab.com/foundrynet/foundryvtt/-/issues/5704

In the meantime, you can build an image locally from the branch using:

```console
docker build \
  --build-arg VERSION=9.220 \
  --tag felddy/foundryvtt:test \
  https://github.com/felddy/foundryvtt-docker.git#release/9.220
```

**Comment by felddy** [comment]
Posting [the reply from Cory](https://gitlab.com/foundrynet/foundryvtt/-/issues/5704#note_646743272) here in case it is deleted from GitLab: 

> 

> Hey there @felddy!
> Due to some of our build tooling that is based on NPM, we strictly require a SemVer compatible `version` in `package.json`. This value is not used inside of Foundry, and could effectively be `0.0.0`
> I would recommend your script join release.generation and `release.build` such as `9.220`, but you could also use just `release.build`

**Comment by felddy** [comment]
The branch is now passing all the tests.  A test container is published from the branch.

You can pull the image using: `felddy/foundryvtt:release-9.220` 

I'm going to leave this issue open as I still need to figure out how tagging will work for pre-releases and update the documentation accordingly.

**Comment by caewok** [comment]
Great; Thanks!


Sent from Yahoo Mail for iPhone


On Monday, August 9, 2021, 10:36, Mark Feldhousen ***@***.***> wrote:




The branch is now passing all the tests. A test container is published from the branch.

You can pull the image using: felddy/foundryvtt:release-9.220

I'm going to leave this issue open as I still need to figure out how tagging will work for pre-releases and update the documentation accordingly.

**Comment by caewok** [comment]
FYI, I had a chance to test, and seems like it is working now. Thanks again!

**Comment by caewok** [comment]
Update for those who may be interested: I updated to V9 Prototype 2 by adding it to container_caches and adding FOUNDRY_VERSION=0.9.224 to my docker compose file.

**Comment by felddy** [comment]
`9.224` was released.  I have update (and renamed) the branch with the prototype support. 

The `--noupdate` flag is no longer a blocker, but there is still something funky going on with updates in general. 

You can pull the image using: `felddy/foundryvtt:prerelease-9.x`

I still need to do more testing, and determine how to handle the new versioning scheme.  I am leaning towards having only `:release` and `:pre-release` alias tags in addition to the normal semver tags.

**Comment by caewok** [comment]
That sounds like a good plan. I would use the pre-release tag. I usually run one docker image just for personal testing of modules and such on whatever the latest pre-release is. Then another for actual games on the stable version.

**Comment by felddy** [comment]
I've updated the `felddy/foundryvtt:prerelease-9.x` image to support FoundryVTT `9.225`

See: https://foundryvtt.com/releases/9.225

**Comment by felddy** [comment]
The funkiness with `--noupdate` flag is still occurring.  I've opened an issue on the FoundryVTT Gitlab repo.

See: https://gitlab.com/foundrynet/foundryvtt/-/issues/6012

**Comment by sPOiDar** [comment]
I built the container with 9.226 and had no problems.

Also, 9.225 downloads from foundry upstream now appear to fail.

**Comment by felddy** [comment]
I've updated the felddy/foundryvtt:prerelease-9.x image to support FoundryVTT 9.226

See: https://foundryvtt.com/releases/9.226

The [`--noupdate` issue](https://gitlab.com/foundrynet/foundryvtt/-/issues/6012) I opened was fixed, but... The interface doesn't seem to be pulling the UI text from the localization file.  So the Docker-specific warnings that the entry point patches in are not being displayed.  Not a showstopper but something I'll need to track down.
